staging/src/k8s.io/dynamic-resource-allocation/cel/compile.go

```diff
@@ -122,18 +130,35 @@ func (c compiler) CompileCELExpression(expression string, envType environment.Ty
 		return resultError("unexpected compilation error: "+err.Error(), apiservercel.ErrorTypeInternal)
 	}
 	prog, err := env.Program(ast,
+		// cel.CostLimit is also set to the VAP PerCallLimit as part of
+		// the base environment.
+		//
+		// This call here should override that. In practice it shouldn't
+		// matter because the limits are the same.
+		cel.CostLimit(resourceapi.CELSelectorExpressionMaxCost),
```

Comment:
Why we want to overwrite it if the limit is the same?
FYI the limit is not set only for VAP but currently for all cel expression used in Kubernetes(VAP, CRD validation rules, webhook, authz, etc)

Reply:
Because it's not guaranteed to be the same. We might decide to use a different limit for DRA.

Different usages for CEL have different timing requirements and thus should decide individually about their cost limits. I find it odd that the shared environment package sets a limit that is defined for VAP.

Reply:
Wouldn't it be better to reuse the same limit until we decided to have a different limit for DRA?

Reply:
Why? They are conceptually different. One is for VAP, the other for DRA. Reusing the VAP constant would give the wrong impression that they must be the same.

Reply:
To make it more obvious that the code is not required to use that limit in particular, I'm now using it merely as default for an `Options` field. It's still a constant in the DRA API.